### PR TITLE
Add Public feed

### PR DESCRIPTION
**Pull request by pabloemilio97**
Implements public feed basic functionalities

- Add new servlet: MessageFeedServlet.java
- Add method to Datastore.java: getAllMessages
- Encapsulate common code between getAllMessages and getMessages in Datastore.java
- Modify MessageFeedServlet.java to use Datastore information in doGet method

### CODE REVIEW

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
-// used for using FetchOptions 
+// used for using FetchOptions
```
Comment (gauravmishra) [inline src/main/java/com/google/codeu/data/Datastore.java]: the import seems self explanatory...you can remove this comment here.

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package com.google.codeu.servlets;
```
Comment (gauravmishra) [inline src/main/java/com/google/codeu/servlets/MessageFeedServlet.java:1]: This pull requests contains mostly datastore changes. Hence you should ideally make the servlet changes a separate pull request for modularity.

**Review by gauravmishra — COMMENTED** [review]
Nice!

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java (hunk 2 above)
Comment (pabloemilio97) [reply src/main/java/com/google/codeu/servlets/MessageFeedServlet.java:1]: Sure! I have a question: should the PR be separated although this branch is the "main" feature branch for public feed? Since all modular features have been merged into this one.

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (pabloemilio97) [reply src/main/java/com/google/codeu/data/Datastore.java]: This and the function getTotalMessageCount come from a change made by @AnnaKawakami , they're here because I rebased master in this branch. Should I still make these changes?

Comment (AnnaKawakami) [reply src/main/java/com/google/codeu/data/Datastore.java]: You can delete the self-explanatory comment! Also, did you mean this this the new master branch? I can remove the function if so

**Review by AnnaKawakami — APPROVED** [review]
Looks good! :)

File: src/main/java/com/google/codeu/data/Datastore.java (hunk 1 above)
Comment (pabloemilio97) [reply src/main/java/com/google/codeu/data/Datastore.java]: No, this is not a new master branch but I added things from master for it to not have any conflicts! What do you think? @gauravmishra
